scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1156,33 +1156,51 @@
              ;; Return all 0's for a user who does NOT have any running job.
              (zipmap (util/get-all-resource-types db) (repeat 0.0)))})))
 
+(defn limit-over-quota-jobs
+  "Filters task-ents, preserving at most (config/max-over-quota-jobs) that would exceed the user's quota"
+  [task-ents quota]
+  (let [over-quota-job-limit (config/max-over-quota-jobs)]
+    (->> task-ents
+         (map (fn [task-ent] [task-ent (util/job->usage (:job/_instance task-ent))]))
+         (reductions (fn [[prev-task total-usage] [task-ent usage]] [task-ent (merge-with + usage total-usage)]))
```

Comment:
Nitpick: should we merge the two reductions to avoid the intermediate vector generation?

Reply:
Done